react/features/recent-list/components/RecentList.native.js

```diff
@@ -1,202 +1,241 @@
+     * Creates a displayable list item of a recent list entry.
      *
-     * @param {string} serverName - The recent list entry being rendered.
      * @private
+     * @param {Object} item - The recent list entry.
      * @returns {Object}
      */
-    _getColorForServerName(serverName) {
-        let nameHash = 0;
-
-        for (let i = 0; i < serverName.length; i++) {
-            nameHash += serverName.codePointAt(i);
-        }
-
-        return styles[`avatarRemoteServer${(nameHash % 5) + 1}`];
+    _toDisplayableItem(item) {
+        const { _homeServer } = this.props;
+        const location = parseURIString(item.conference);
+        const baseURL = `${location.protocol}//${location.host}`;
+        const serverName = baseURL === _homeServer ? null : location.host;
+
+        return {
+            colorBase: serverName,
+            key: `key-${item.conference}-${item.date}`,
+            lines: [
+                this._toDateString(item.date),
+                this._toDurationString(item.conferenceDuration),
+                serverName
+            ],
+            title: location.room,
+            url: item.conference
+        };
     }
 
+    _toDisplayableList: () => Array<Object>
+
     /**
-     * Renders the conference duration if available.
+     * Transforms the history list to a displayable list
+     * with sections.
      *
-     * @param {Object} recentListEntry - The recent list entry being rendered.
      * @private
-     * @returns {ReactElement}
+     * @returns {Array<Object>}
      */
-    _renderConfDuration({ durationString }) {
-        if (durationString) {
-            return (
-                <View style = { styles.infoWithIcon } >
-                    <Icon
-                        name = 'timer'
-                        style = { styles.inlineIcon } />
-                    <Text style = { styles.confLength }>
-                        { durationString }
-                    </Text>
-                </View>
-            );
+    _toDisplayableList() {
+        const { _recentList, t } = this.props;
+        const todaySection = NavigateSectionList.createSection(
+            t('recentList.today'),
+            'today'
+        );
+        const yesterdaySection = NavigateSectionList.createSection(
+            t('recentList.yesterday'),
+            'yesterday'
+        );
+        const earlierSection = NavigateSectionList.createSection(
+            t('recentList.earlier'),
+            'earlier'
+        );
+        const today = new Date().toDateString();
+        const yesterdayDate = new Date();
+
+        yesterdayDate.setDate(yesterdayDate.getDate() - 1);
+
+        const yesterday = yesterdayDate.toDateString();
+
+        for (const item of _recentList) {
+            const itemDay = new Date(item.date).toDateString();
+            const displayableItem = this._toDisplayableItem(item);
+
+            if (itemDay === today) {
+                // the item is today
```

Comment:
I admittedly live in the school of thought that comments should be avoided when possible, especially if you can make the code itself express what is being done. I'm fairly certain this is a case where you have more than adequately conveyed that the item's date is equal to today.